src/hooks/useTally.js

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from "react"
+import { useSelector } from "react-redux"
+import useReports from "./useReports"
+
+export default function useTally() {
+    const {report} = useReports()
+    console.log(report)
+    
+    const [firstCounter, setFirstCounter] = useState(0)
+    const [secondCounter, setSecondCounter] = useState(0)
+    const [thirdCounter, setThirdCounter] = useState(0)
+
+    const today = useSelector(state => state.todayDate)
+    const fourWeeks = useSelector(state => state.fourWeeks)
+    const eightWeeks = useSelector(state => state.eightWeeks)
+    const twelveWeeks = useSelector(state => state.twelveWeeks)
+    
+    useEffect(() => {  
+        const moment = require("moment")
+        let newToday = today.slice(0, -15)
+        let newFourWeeks = fourWeeks.slice(0, -15)
+        let newEightWeeks = eightWeeks.slice(0, -15)
+        let newTwelveWeeks = twelveWeeks.slice(0, -15)
+         
+      
+          for (let i = 0; i < report.length; i++) {
+            if (moment(report[i]).isBetween(newFourWeeks, newToday, undefined, '[]') === true) {
+                  setFirstCounter(firstCounter + 1)
+                  console.log(firstCounter)
+              } 
+            if (moment(report[i]).isBetween(newEightWeeks, newFourWeeks, undefined, '[]') === true) {
+                  setSecondCounter(secondCounter + 1)
+                  console.log(secondCounter)
+              } 
+            if (moment(report[i]).isBetween(newTwelveWeeks, newEightWeeks, undefined, '[]') === true) {
+                  setThirdCounter(thirdCounter + 1)
+                  console.log(thirdCounter)
+              } 
+          }
```

Comment:
Please read this: https://reactjs.org/docs/react-component.html#setstate
TL;DR:
> Basically, don't call setState in a loop. What's happening here is exactly what the docs are referring to: this.state is returning the previous value, as the **pending state update has not been applied yet**.

Try to use a counter variable and save the endCount with the `setFunction`.
```suggestion
          let counterFirst = 0
          let counterSecond = 0
          let counterThird = 0
          
          for (let i = 0; i < report.length; i++) {
            if (moment(report[i]).isBetween(newFourWeeks, newToday, undefined, '[]') === true) {
                  counterFirst++
              } 
            if (moment(report[i]).isBetween(newEightWeeks, newFourWeeks, undefined, '[]') === true) {
                 counterSecond++
              } 
            if (moment(report[i]).isBetween(newTwelveWeeks, newEightWeeks, undefined, '[]') === true) {
                  counterThird++
              } 
          }
          setFirstCounter(counterFirst)
          setSecondCounter(counterSecond)
          setThirdCounter(counterThird)
```

Reply:
Very good idea! Will have a look see if this fixes it